### ISSUE TITLE
http://mjpeg.sanford.io/count.mjpeg

### ISSUE THREAD

**Issue by Nibbler999** [mmaelzer/mjpeg-consumer#11]
I noticed the module doesn't work properly with this stream. It outputs files which contain two images.

``` javascript
var request = require("request");
var MjpegConsumer = require("mjpeg-consumer");
var FileOnWrite = require("file-on-write");

var writer = new FileOnWrite({ 
    path: './video',
    ext: '.jpg'
});
var consumer = new MjpegConsumer();

request("http://mjpeg.sanford.io/count.mjpeg").pipe(consumer).pipe(writer);
```

**Comment by mmaelzer** [comment]
Hey Tom, thanks for the feedback, as always. You're right, there's definitely an issue. After spending some time revisiting this library, I realize my approach should probably be rethought. I'll try to clean things up over the coming holiday weekend.

**Comment by mmaelzer** [comment]
Alright @Nibbler999, this is fixed as of 9764f7dad9b3d70c. I've also added tests and code coverage while fixing this issue (long overdue, honestly). That URL was extremely useful for testing.

**Comment by Nibbler999** [comment]
Thanks for that, the new version is certainly much simpler. Unfortunately there are several bugs that prevent it working properly in my application. I'll send in some PRs next week.
